modules/core/04-channel/keeper/packet.go

```diff
@@ -133,10 +133,11 @@ func (k Keeper) RecvPacket(
 		return errorsmod.Wrap(types.ErrChannelNotFound, packet.GetDestChannel())
 	}
 
-	if channel.State != types.OPEN {
+	counterpartyLastPacketSent, _ := k.GetCounterpartyLastPacketSequence(ctx, packet.GetDestPort(), packet.GetDestChannel())
+	if channel.State != types.OPEN && (channel.FlushStatus != types.FLUSHING || counterpartyLastPacketSent > packet.GetSequence()) {
 		return errorsmod.Wrapf(
 			types.ErrInvalidChannelState,
-			"channel state is not OPEN (got %s)", channel.State.String(),
+			"packets cannot be received on channel with state (%s) and flush status (%s)", channel.State, channel.FlushStatus,
```

Comment:
The error message will not provide enough information if the channel is not open and flushing, but the packet sequence is > `counterpartyLastPacketSent`. I guess it would be very informative if we have different error messages depending on the condition that was not satisfied?